packages/wrangler/src/autoconfig/run.ts

```diff
@@ -120,3 +168,99 @@ export async function runAutoConfig(
 
 	return;
 }
+
+function usesTypescript(projectPath: string) {
+	return hasTsConfig(projectPath);
+}
+
+function hasTsConfig(path: string) {
+	return existsSync(join(`${path}`, `tsconfig.json`));
+}
+
+type PackageToInstall = {
+	package: string;
+	depType: "dependency" | "devDependency";
+};
+
+type ScriptToAdd = {
+	name: string;
+	value: string;
+};
+
+export async function buildAndConfirmOperationsSummary(
+	autoConfigDetails: AutoConfigDetails,
+	wranglerConfigToWrite: RawConfig
+): Promise<{
+	confirmed: boolean;
+	packagesToInstall: PackageToInstall[];
+	scriptsToAdd: ScriptToAdd[];
+}> {
+	const packagesToInstall: PackageToInstall[] = [];
+	const scriptsToAdd: ScriptToAdd[] = [];
+	if (autoConfigDetails.packageJson) {
+		const devDependencies = autoConfigDetails.packageJson.devDependencies ?? {};
+		const shouldInstallLatestWrangler =
+			!("wrangler" in devDependencies) ||
+			(typeof devDependencies["wrangler"] === "string" &&
+				!devDependencies["wrangler"].startsWith("^"));
```

Comment:
Presumably this can only be `string` or `undefined`, so this could be simplified to `!devDependencies.wrangler || !devDependencies.wrangler.startsWith("^")`?

Reply:
> Presumably this can only be `string` or `undefined`, so this could be simplified to `!devDependencies.wrangler || !devDependencies.wrangler.startsWith("^")`?

No that wouldn't be necessary and we'd need a type cast there since `devDependencies.wrangler` would be `unknown`

@penalosa do you know if there was a specific reason why `unknown` was used here?
https://github.com/cloudflare/workers-sdk/blob/52863090c3cb97d3cee10ede84420018305b6354/packages/workers-utils/src/parse.ts#L127